### PR TITLE
tap: Support `tls` labeling

### PR DESCRIPTION
**Pull request by olix0r**
The proxy's metrics are instrumented with a `tls` label that describes
the state of TLS for each connection and associated messges.

This same level of detail is useful to get in `tap` output as well.

This change updates Tap in the following ways:
* `TapEvent`
  * Added `source_meta` field including source labels
  * `proxy_direction` enum indicates which proxy server was used.
* The proxy adds a `tls` label to both source and destination meta indicating the state of each peer's connection
* The CLI uses the `proxy_direction` field to determine which `tls` label should be rendered.

### CODE REVIEW

**Comment by olix0r** [comment]
I think this isn't _quite_ right -- should have a follow-on today

File: proto/common.proto [hunk 1]
```diff
@@ -68,10 +68,18 @@ message Eos {
...
 message TapEvent {
   TcpAddress source = 1;
+  EndpointMeta source_meta = 5;
 
   TcpAddress   destination      = 2;
   EndpointMeta destination_meta = 4;
 
+  ProxyDirection proxy_direction = 6;
+  enum ProxyDirection {
+    UNKNOWN = 0;
```
Comment (olix0r) [inline proto/common.proto]: Note: UNKNOWN is used as the default so that we don't misinterpret events where this field is unset.

**Comment by hawkw** [comment]
@olix0r 
> I think this isn't _quite_ right -- should have a follow-on today

Is this branch currently ready for review?

**Comment by olix0r** [comment]
yes

**Comment by olix0r** [comment]
published to docker as git-c29c52e9

File: proxy/src/control/pb.rs [hunk 2]
```diff
@@ -205,3 +190,37 @@ impl<'a> TryFrom<&'a Event> for common::TapEvent {
         Ok(tap_ev)
     }
 }
+
+impl ctx::transport::Server {
```
Comment (hawkw) [inline proxy/src/control/pb.rs]: I initially thought the impls on types defined in a different module seemed a big confusing, but given that we wouldn't expect to use these methods anywhere outside of `tap`, I think this makes sense in context 👍

**Pull request merged**